app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SessionsController < Devise::SessionsController
+  layout 'empty', :only => [ :timeout_check ]
+
+  def timeout_check
+    return if !session || !session["warden.user.user.session"] # session["warden.user.user.session"] comes from adding timeoutable to the user model
```

Comment:
What value is stored "warden.user.user.session". Can we change from the name of this key?

Reply:
It's the default from Warden.  I'm thinking we should keep it the same, because a bunch of things may depend upon it now or in the future.

Reply:
Ok I added this to the `timeout_check` method in sessions_controller.rb:
```
# session["warden.user.user.session"] comes from adding timeoutable to the user model
# example: {"last_request_at"=>1537283353}
```